builtin/providers/aws/resource_aws_route53_zone.go

```diff
@@ -104,3 +111,20 @@ func resourceAwsRoute53Wait(r53 *route53.Route53, ref string) (result interface{
 	}
 	return true, status, nil
 }
+
+func resourceAwsGoRoute53Wait(r53 *awsr53.Route53, ref *awsr53.GetChangeRequest) (result interface{}, state string, err error) {
+
+	status, err := r53.GetChange(ref)
+	if err != nil {
+		return nil, "UNKNOWN", err
+	}
+	return true, *status.ChangeInfo.Status, nil
+}
+
+// CleanChangeID is used to remove the leading /change/
+func CleanChangeID(ID string) string {
```

Comment:
I'm okay with putting it here at the bottom of the file. Let's add unit tests for it too. 

Also, I think we should not export it (lowercase the names). Same with the above ones too.